src/foam/nanos/auth/User.js

```diff
@@ -158,6 +162,18 @@ foam.CLASS({
       containsPII: false,
       documentation: 'The username of the User.',
       section: 'userInformation',
+      validationPredicates: [
+        {
+          args: ['userName', 'type'],
+          predicateFactory: function(e) {
+            return e.OR(
+              e.NEQ(foam.nanos.auth.User.TYPE, 'User'),
```

Comment:
is this NOT correct?

Reply:
This predicate only returns false (throws the error) when the class is User and the user doesn't have a username set.

Reply:
had to stop and think about this - but this works.

This wasn't the natural way I thought about this.. Which is just interesting.
if ( (user = User & userName empty )  ) then return fail
however here you've set
if ( user != User || userName not empty ) then return pass
I had to look up some of the laws of boolean algebra - but the two are equivalent.

Reply:
Yep, it's just the negation of if ( (user = User & userName empty ) ) since predicate factory throws when the predicate is false and not true.